web/concrete/src/Application/Application.php

```diff
@@ -130,6 +130,45 @@ public function clearCaches()
         OpCache::clear();
     }
 
+   /**
+     * Utility method for clearing page caches and block caches that has btCacheBlockOutputClearOnEvents option enabled
+     */
+    public function partialClearCaches()
+    {
+        \Events::dispatch('on_partial_cache_flush');
```

Comment:
I'm just posting a solution for the problem, we can improve then the code if we are ok about the approach.

I saw the PR that improves performance in User class, and i have just a comment about its usability.
I think it would be better if we find more elegant way to call these objects or at least a simple method like Application::getDatabase() and Application::getDirector()